core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -431,6 +433,26 @@ class AdminManager(val config: KafkaConfig,
                 (name, value) => new DescribeConfigsResponseData.DescribeConfigsResourceResult().setName(name)
                   .setValue(value.toString).setConfigSource(ConfigSource.DYNAMIC_BROKER_LOGGER_CONFIG.id)
                   .setIsSensitive(false).setReadOnly(false).setSynonyms(List.empty.asJava))
+          case ConfigResource.Type.CLIENT =>
+            val clientId = resource.resourceName
+            val defaultProps = adminZkClient.fetchEntityConfig(ConfigType.Client, ConfigEntityName.Default)
+            val clientProps = adminZkClient.fetchEntityConfig(ConfigType.Client, if (clientId.isEmpty) ConfigEntityName.Default else clientId)
+            val overlayedProps = new Properties()
+            overlayedProps.putAll(defaultProps)
+            overlayedProps.putAll(clientProps)
+            val configMap = overlayedProps.stringPropertyNames.asScala
+              .filter(ClientConfigs.isClientConfig).map{key => (key -> overlayedProps.getProperty(key))}.toMap
```

Comment:
Java's `Properties` is a `HashTable`. This can be done by using `flatMap`. E.g.:

```scala
overlayedProps.asScala.flatMap { (key, value) =>
  val key = key.toString
  val value = value.toString

  if (ClientConfigs.isClientConfig(key) {
    None
  } else {
    Some(key -> value)
  }
}

Reply:
if (ClientConfigs.isClientConfig(key) {
    Some(key -> value)
  } else {
    None
  }